src/components/font_size_editor/font_size_editor.ts

```diff
@@ -0,0 +1,109 @@
+import { Component, useExternalListener, useRef, useState } from "@odoo/owl";
+import { DEFAULT_FONT_SIZE, FONT_SIZES } from "../../constants";
+import { clip } from "../../helpers/index";
+import { setStyle } from "../../registries/index";
+import { SpreadsheetChildEnv } from "../../types/index";
+import { css } from "../helpers/css";
+import { isChildEvent } from "../helpers/dom_helpers";
+
+interface State {
+  isOpen: boolean;
+}
+
+interface Props {
+  onDone: () => void;
+  dropdownStyle: string;
+}
+
+// -----------------------------------------------------------------------------
+// TopBar
+// -----------------------------------------------------------------------------
+css/* scss */ `
+  .o-font-size-editor {
+    input.o-font-size {
+      height: 20px;
+      width: 23px;
+    }
+    input[type="number"] {
+      -moz-appearance: textfield;
+    }
+    input::-webkit-outer-spin-button,
+    input::-webkit-inner-spin-button {
+      -webkit-appearance: none;
+    }
+  }
+`;
+
+export class FontSizeEditor extends Component<Props, SpreadsheetChildEnv> {
+  static template = "o-spreadsheet-FontSizeEditor";
+  static components = {};
+  fontSizes = FONT_SIZES;
+
+  dropdown: State = useState({ isOpen: false });
+  openedEl: HTMLElement | null = null;
+
+  private inputRef = useRef("inputFontSize");
+  private rootEditorRef = useRef("FontSizeEditor");
+
+  setup() {
+    useExternalListener(window, "click", this.onExternalClick);
```

Comment:
```suggestion
    useExternalListener(window, "click", this.onExternalClick, { capture: true });
```
Thanks to that, if you click on an element which have a `stopPropagation` on any itself or any parent element, it will still close the dropdown